controllers/snapshot/snapshot_adapter.go

```diff
@@ -81,6 +82,11 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (reconciler.Operation
 			"IntegrationTestScenarios", len(*integrationTestScenarios))
 		for _, integrationTestScenario := range *integrationTestScenarios {
 			integrationTestScenario := integrationTestScenario //G601
+			if !reflect.ValueOf(integrationTestScenario.Spec.Environment).IsZero() {
```

Comment:
I *think* you can just do `if integrationTestScenario.Spec.Environment != ""`

Reply:
The value of Environment is a struct, [TestEnvironment](https://github.com/redhat-appstudio/integration-service/blob/main/api/v1alpha1/integrationtestscenario_types.go#L39), not a string, so that `!= ""` doesn't work.